### PR TITLE
IZE-479 New default state bucket name

### PR DESCRIPTION
**Pull request by psihachina**
## Changelog:
- new default state bucket name

## Test:
[asciicast](https://asciinema.org/a/Z3oFDHvyYFrbAp3nC9U1vp3SZ)

### CODE REVIEW

File: internal/commands/gen/tfenv.go [hunk 1]
```diff
@@ -184,3 +203,22 @@ func GenerateTerraformFiles(project *config.Project, terraformStateBucketName st
 
 	return nil
 }
+
+func checkTFState(sess *session.Session, name string) bool {
```
Comment (AutomationD) [inline internal/commands/gen/tfenv.go:207]: This function seems to be checking the existence of a lock but not S3 bucket.

The goal is to check if bucket exists (old code) and use that. If the bucket doesn't exist - use new defaults.
If it is set via parameters then use that value.

Does it make sense?

**Review by AutomationD — CHANGES_REQUESTED** [review]

**Comment by psihachina** [comment]
@AutomationD 
[asciicast](https://asciinema.org/a/eIi3QLvnawvZbiLuqirE69GOC)

File: internal/commands/gen/tfenv.go [hunk 2]
```diff
@@ -99,7 +106,19 @@ func GenerateTerraformFiles(project *config.Project, terraformStateBucketName st
 	}
 
 	if len(tf.StateBucketName) == 0 {
-		tf.StateBucketName = fmt.Sprintf("%s-tf-state", project.Namespace)
+		exist := checkTFState(project.Session, fmt.Sprintf("%s-tf-state", project.Namespace))
```
Comment (AutomationD) [inline internal/commands/gen/tfenv.go:109]:
```suggestion
		legacyBucketExists := checkTFStateBucket(project.Session, fmt.Sprintf("%s-tf-state", project.Namespace))
```

File: internal/commands/gen/tfenv.go [hunk 3]
```diff
@@ -99,7 +106,19 @@ func GenerateTerraformFiles(project *config.Project, terraformStateBucketName st
...
+		if exist {
```
Comment (AutomationD) [inline internal/commands/gen/tfenv.go:110]:
```suggestion
		if legacyBucketExists {
```

File: internal/commands/gen/tfenv.go [hunk 4]
```diff
@@ -184,3 +203,21 @@ func GenerateTerraformFiles(project *config.Project, terraformStateBucketName st
 
 	return nil
 }
+
+func checkTFState(sess *session.Session, name string) bool {
```
Comment (AutomationD) [inline internal/commands/gen/tfenv.go:207]:
```suggestion
func checkTFStateBucket(sess *session.Session, name string) bool {
```

File: internal/commands/gen/tfenv.go [hunk 5]
```diff
@@ -99,7 +106,19 @@ func GenerateTerraformFiles(project *config.Project, terraformStateBucketName st
 	}
 
 	if len(tf.StateBucketName) == 0 {
-		tf.StateBucketName = fmt.Sprintf("%s-tf-state", project.Namespace)
+		exist := checkTFState(project.Session, fmt.Sprintf("%s-tf-state", project.Namespace))
+		if exist {
+			tf.StateBucketName = fmt.Sprintf("%s-tf-state", project.Namespace)
```
Comment (AutomationD) [inline internal/commands/gen/tfenv.go:111]:
```suggestion
			// If we found an existing bucket that conforms with the legacy format use it.
			tf.StateBucketName = fmt.Sprintf("%s-tf-state", project.Namespace)
```

File: internal/commands/gen/tfenv.go [hunk 6]
```diff
@@ -99,7 +106,19 @@ func GenerateTerraformFiles(project *config.Project, terraformStateBucketName st
...
+		} else {
+			resp, err := sts.New(project.Session).GetCallerIdentity(
+				&sts.GetCallerIdentityInput{},
+			)
+			if err != nil {
+				return err
+			}
+
+			tf.StateBucketName = fmt.Sprintf("%s-%s-tf-state", project.Namespace, *resp.Account)
```
Comment (AutomationD) [inline internal/commands/gen/tfenv.go:120]:
```suggestion
			// If we haven't found an existing legacy format state bucket use a <NAMESPACE>-<AWS_ACCOUNT>-tf-state bucket as default (unless overriden with other parameters.
			tf.StateBucketName = fmt.Sprintf("%s-%s-tf-state", project.Namespace, *resp.Account)
```

**Review by AutomationD — CHANGES_REQUESTED** [review]

**Review by AutomationD — APPROVED** [review]
Looks good, thank you!

**Pull request merged**